app/src/main/java/com/techcourse/handler/HandlerAdapterRegistry.java

```diff
@@ -0,0 +1,27 @@
+package com.techcourse.handler;
+
+import com.interface21.webmvc.servlet.HandlerAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class HandlerAdapterRegistry {
+
+    private final List<HandlerAdapter> handlerAdapters;
+
+    public HandlerAdapterRegistry() {
+        this.handlerAdapters = new ArrayList<>();
+    }
+
+    public void addHandlerAdapter(HandlerAdapter handlerAdapter) {
+        handlerAdapters.add(handlerAdapter);
+    }
+
+    public HandlerAdapter getHandlerAdapter(Object handler) {
```

Comment:
예외를 던진다면 시그니처에 명시하는 게 호출하는 쪽에서 예외 파악에 도움이 될 것 같아요.

```suggestion
    public HandlerAdapter getHandlerAdapter(Object handler) throws IllegalArgumentException {
```